product_customer_code/models/product.py

```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+
+###########################################################################
+#    Module Writen to OpenERP, Open Source Management Solution
+#
+#    Copyright (c) 2012 Vauxoo - http://www.vauxoo.com
+#    All Rights Reserved.
+#    info@vauxoo.com
+############################################################################
+#    Coded by: Rodo (rodo@vauxoo.com),Moy (moylop260@vauxoo.com)
+############################################################################
+
+from odoo import models, fields, api
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    product_customer_code_ids = fields.One2many(
+        comodel_name='product.customer.code',
+        inverse_name='product_id',
+        string='Customer Codes',
+        copy=False
+    )
+
+    @api.model
+    def name_search(self, name='', args=None, operator='ilike', limit=100):
+        res = super(ProductProduct, self).name_search(
+            name=name, args=args, operator=operator, limit=limit)
+        if not res:
+            partner_id = self._context.get('partner_id')
+            if partner_id:
+                product_customer_code_obj = self.env['product.customer.code']
+                product_codes = product_customer_code_obj.search([
+                    ('product_code', '=', name),
```

Comment:
IMO these are the types of questions that need to be asked during migration, because this is the point where we have the most flexibility in the way that a module operates. Once we release stable, changing things becomes a lot more difficult.

I totally agree with using `operator` and that we should eval for equal length of res instead of just an either or. For the little bit of effort this would take, this would increase the functionality of this module quite a bit outside of its intended - which is good IMO as long as we don't sacrifice anything in the process.

I think that we may be getting a bit liberal with it by searching children though, because then the module's functionality would be changed drastically without the ability to revert. 

Consider a giant conglomerate with multiple business entities under it. These entities could all have separate product codes, but we would get an aggregate if searching the children without a way to filter away from these results. 

On the flip-side, current operation we just have to search more codes if we wanted some sort of aggregate report. I could also be misunderstanding this side of the proposal though, so please let me know if I was just talking stupid 😉